drivers/claims_reporting/app/models/claims_reporting/engagement_trends.rb

```diff
@@ -170,54 +161,130 @@ def detail_row_headers
     end
 
     def cohorts
+      return engaged_history_cohorts if filter.cohort_type == :engaged_history
+
+      selected_period_cohorts
+    end
+
+    private def cohort_scope(engaged_for_days)
+      scope = ClaimsReporting::MemberEnrollmentRoster.engaged_for(engaged_for_days)
+      # NOTE: these all expect scope to be something that can be joined to client so we'll
+      # use data_source, since everyone has to have one of those
+      # They also expect @filter to be set
+      @filter = filter
+
+      # apply filters via warehouse clients... This
+      # is the only place linking this to warehouse_db
+      unless Rails.env.development?
+        hmis_scope = ::GrdaWarehouse::DataSource.destination.joins(:clients)
+        hmis_scope = filter_for_age_ranges(hmis_scope) if filter.age_ranges.present?
+        hmis_scope = filter_for_gender(hmis_scope) if filter.genders.present?
+        hmis_scope = filter_for_race(hmis_scope) if filter.races.present?
+        hmis_scope = filter_for_ethnicity(hmis_scope) if filter.ethnicities.present?
+        client_ids = hmis_scope.pluck(c_t[:id])
+        scope = scope.joins(:patient).merge(::Health::Patient.where(client_id: client_ids)) if client_ids.any?
+        scope = scope.joins(patient: :patient_referral).merge(::Health::PatientReferral.at_acos(filter.acos)) if filter.acos.present?
+      end
```

Comment:
@amichal , I _think_ this should not cause any issues if you don't set any filters, if it is, we should figure out why.  It should only be limiting with HMIS data if something has been chosen.